tests/net/lib/lwm2m/lwm2m_engine/src/main.c

```diff
@@ -0,0 +1,375 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/fff.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/ztest.h>
+
+#include "lwm2m_engine.h"
+#include "lwm2m_rd_client.h"
+
+#include "stubs.h"
+
+LOG_MODULE_REGISTER(lwm2m_engine_test);
+
+DEFINE_FFF_GLOBALS;
+#define FFF_FAKES_LIST(FAKE)
+
+static uint8_t my_buf[256];
+static struct lwm2m_message my_msg;
+static struct lwm2m_engine_obj_field my_obj_field;
+
+static int lwm2m_get_res_buf_custom_fake(const struct lwm2m_obj_path *path, void **buffer_ptr,
+					 uint16_t *buffer_len, uint16_t *data_len,
+					 uint8_t *data_flags)
+{
+	if (buffer_ptr)
+		*buffer_ptr = my_buf;
+	if (buffer_len)
+		*buffer_len = sizeof(my_buf);
+	if (data_len)
+		*data_len = 1;
+
+	return 0;
+}
+
+static struct lwm2m_message *find_msg_custom_fake(struct coap_pending *pending,
+						  struct coap_reply *reply)
+{
+	return &my_msg;
+}
+
+static struct lwm2m_engine_obj_field *
+lwm2m_get_engine_obj_field_custom_fake(struct lwm2m_engine_obj *obj, int res_id)
+{
+	return &my_obj_field;
+}
+
+static int lwm2m_get_bool_custom_fake(const struct lwm2m_obj_path *path, bool *value)
+{
+	*value = false;
+
+	return 0;
+}
+
+static void test_service(struct k_work *work)
+{
+	LOG_INF("Test service");
+	k_sleep(K_MSEC(10));
+}
+
+static void setup(void)
+{
+	/* Register resets */
+	DO_FOREACH_FAKE(RESET_FAKE);
+
+	/* reset common FFF internal structures */
+	FFF_RESET_HISTORY();
+
+	clear_socket_events();
+	lwm2m_get_res_buf_fake.custom_fake = lwm2m_get_res_buf_custom_fake;
+	find_msg_fake.custom_fake = find_msg_custom_fake;
+	lwm2m_get_engine_obj_field_fake.custom_fake = lwm2m_get_engine_obj_field_custom_fake;
+	lwm2m_get_bool_fake.custom_fake = lwm2m_get_bool_custom_fake;
+}
+
+ZTEST_SUITE(lwm2m_engine, NULL, NULL, NULL, NULL, NULL);
+
+ZTEST(lwm2m_engine, test_start_stop)
+{
+	int ret;
+	struct lwm2m_ctx ctx;
+	char host_name[10] = "my_host";
+
+	setup();
```

Comment:
Instead of calling this for every test case, you could register `before_fn` routine for `ZTEST_SUITE` to be called before each test, and do the setup there.

Reply:
Thanks for the tip. I've made the changes.